### ISSUE TITLE
ChannelShuffle transforms

### ISSUE THREAD

**Issue by dhruvbird** [pytorch/vision#7620]
### 🚀 The feature

Is there any chance torchvision could implement a ChannelShuffle transform?

### Motivation, pitch

Albumenations has it.
https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.ChannelShuffle

### Alternatives

I'm currently implementing this myself by hand or using Albumenations.

### Additional context

Channel Shuffle is useful since it allows me to mix up the input to prevent overfitting.

cc @vfdev-5

**Comment by NicolasHug** [comment]
Sure, that sounds reasonable. Perhaps only on the V2 namespace. Thoughts @vfdev-5 @pmeier ?

**Comment by pmeier** [comment]
`RandomPhotometricDistort` (v2 only) uses that internally already:

https://github.com/pytorch/vision/blob/e5bf7cf85dcab36a313890deb57b8eb864f06292/torchvision/transforms/v2/_color.py#L270-L273

https://github.com/pytorch/vision/blob/e5bf7cf85dcab36a313890deb57b8eb864f06292/torchvision/transforms/v2/_color.py#L244-L257

So basically we only need to extract this out into a transform. That would also make the implementation more clean, since apart from the channel permutation, all other ops that `RandomPhotometricDistort` is calling have a dedicated functional.

**Comment by pmeier** [comment]
@dhruvbird could you please open a separate issue for that to keep the scope narrow?

**Comment by dhruvbird** [comment]
> 

Sure thing! https://github.com/pytorch/vision/issues/7633
